src/odemis/driver/test/actuator_test.py

```diff
@@ -669,6 +669,278 @@ def test_move_abs(self):
+        f = stage.moveRel({"x": -1e-06, "y": -2e-06, "z": -3e-06})
+        f.result()
+        test.assert_pos_almost_equal(stage.position.value, {"x": 0, "y": 0, "z": 0})
+        test.assert_pos_almost_equal(dependency.position.value, {"x": 0, "y": 0, "z": 0})
+
+        # scaling
+        stage = Convert3DStage("conv", "align", {"orig": dependency}, axes=["x", "y", "z"],
+                               scale=(10, 10, 10))
+        test.assert_pos_almost_equal(stage.position.value, {"x": 0, "y": 0, "z": 0})
+        f = stage.moveRel({"x": 1e-06, "y": 2e-06, "z": 3e-06})
+        f.result()
+        test.assert_pos_almost_equal(stage.position.value, {"x": 1e-06, "y": 2e-06, "z": 3e-06})
+        test.assert_pos_almost_equal(dependency.position.value, {"x": 10e-06, "y": 20e-06, "z": 30e-06})
+        f = stage.moveRel({"x": -1e-06, "y": -2e-06, "z": -3e-06})
+        f.result()
+        test.assert_pos_almost_equal(stage.position.value, {"x": 0, "y": 0, "z": 0})
+        test.assert_pos_almost_equal(dependency.position.value, {"x": 0, "y": 0, "z": 0})
+        # only one axis at a time (to check missing axis doesn't do weird move)
+        f = stage.moveRel({"x": 1e-06})
+        f.result()
+        test.assert_pos_almost_equal(stage.position.value, {"x": 1e-06, "y": 0, "z": 0})
+        test.assert_pos_almost_equal(dependency.position.value, {"x": 10e-06, "y": 0, "z": 0})
+        f = stage.moveRel({"y": 2e-06})
+        f.result()
+        test.assert_pos_almost_equal(stage.position.value, {"x": 1e-06, "y": 2e-06, "z": 0})
+        test.assert_pos_almost_equal(dependency.position.value, {"x": 10e-06, "y": 20e-06, "z": 0})
+        f = stage.moveRel({"z": 3e-06})
+        f.result()
+        test.assert_pos_almost_equal(stage.position.value, {"x": 1e-06, "y": 2e-06, "z": 3e-06})
+        test.assert_pos_almost_equal(dependency.position.value, {"x": 10e-06, "y": 20e-06, "z": 30e-06})
+
+        f = stage.moveRel({"x": -1e-06, "y": -2e-06, "z": -3e-06})
+        f.result()
+        test.assert_pos_almost_equal(stage.position.value, {"x": 0, "y": 0, "z": 0})
+        test.assert_pos_almost_equal(dependency.position.value, {"x": 0, "y": 0, "z": 0})
+
+        # rotation
+        stage = Convert3DStage("conv", "align", {"orig": dependency}, axes=["x", "y", "z"],
+                               rotation=(math.pi / 2, 0, 0))
+        test.assert_pos_almost_equal(stage.position.value, {"x": 0, "y": 0, "z": 0})
+        f = stage.moveRel({"x": 1e-06, "y": 2e-06, "z": 0})
+        f.result()
+        self.assertEqual(stage.position.value, {"x": 1e-06, "y": 2e-06, "z": 0})
+        test.assert_pos_almost_equal(dependency.position.value, {"x": -2e-06, "y": 1e-06, "z": 0})
+        f = stage.moveRel({"x": -1e-06, "y": -2e-06})
+        f.result()
+        test.assert_pos_almost_equal(stage.position.value, {"x": 0, "y": 0, "z": 0})
+        test.assert_pos_almost_equal(dependency.position.value, {"x": 0, "y": 0, "z": 0})
+
```

Comment:
That's fine, but could you add another one testing another dimension, to check it does the right thing also in "3D"? At least, just ry or rx. Ideally, you could check multiple axes...

Reply:
There are tests in `test_ab_rotation` for the 3 angles. I have also added them on `test_move_rel`.